lib/rspec/snapshot/utils.rb

```diff
@@ -0,0 +1,75 @@
+require 'fileutils'
+require 'oj'
+
+module RSpec
+  module Snapshot
+    # Common utilities for snapshots
+    # Most of the work is computing the right filename and interacting with the filesystem
+    module Utils
+      class ConfigError < StandardError; end
+
+      def self.write_snapshot(snap_path, value)
+        RSpec.configuration.reporter.message "Writing snapshot at #{snap_path}"
+        FileUtils.mkdir_p(File.dirname(snap_path)) unless Dir.exist?(File.dirname(snap_path))
+        File.open(snap_path, 'w+') do |file|
+          file.write(serialize(value))
+        end
+      end
+
+      def self.serialize(value, serializer: nil)
+        if serializer
```

Comment:
This method 'supports' a custom serializer, if one could be configured / passed in